sdk/src/jumbf_io.rs

```diff
@@ -110,52 +125,33 @@ pub fn save_jumbf_to_memory(
     Ok(stream.into_inner())
 }
 
-pub fn get_assetio_handler(ext: &str) -> Option<Box<dyn AssetIO>> {
+pub fn get_assetio_handler(ext: &str) -> Option<&dyn AssetIO> {
     let ext = ext.to_lowercase();
-    match ext.as_ref() {
-        "c2pa" => Some(Box::new(C2paIO {})),
-        "jpg" | "jpeg" => Some(Box::new(JpegIO {})),
-        "png" => Some(Box::new(PngIO {})),
-        "mp4" | "m4a" | "mov" if cfg!(feature = "bmff") => Some(Box::new(BmffIO::new(&ext))),
-        "tif" | "tiff" | "dng" => Some(Box::new(TiffIO {})),
-        _ => None,
+
+    if let Some(h) = ASSET_HANDLERS.get(&ext) {
+        Some(h.as_ref())
+    } else {
+        None
     }
 }
 
-pub fn get_cailoader_handler(asset_type: &str) -> Option<Box<dyn CAILoader>> {
+pub fn get_cailoader_handler(asset_type: &str) -> Option<&dyn CAIReader> {
     let asset_type = asset_type.to_lowercase();
-    match asset_type.as_ref() {
-        "c2pa" | "application/c2pa" | "application/x-c2pa-manifest-store" => {
-            Some(Box::new(C2paIO {}))
-        }
-        "jpg" | "jpeg" | "image/jpeg" => Some(Box::new(JpegIO {})),
-        "png" | "image/png" => Some(Box::new(PngIO {})),
-        "avif" | "heif" | "heic" | "mp4" | "m4a" | "application/mp4" | "audio/mp4"
-        | "image/avif" | "image/heic" | "image/heif" | "video/mp4"
-            if cfg!(feature = "bmff") && !cfg!(target_arch = "wasm32") =>
-        {
-            Some(Box::new(BmffIO::new(&asset_type)))
-        }
-        "tif" | "tiff" | "dng" => Some(Box::new(TiffIO {})),
-        _ => None,
+
+    if let Some(h) = ASSET_HANDLERS.get(&asset_type) {
```

Comment:
This could probably be simplified to `ASSET_HANDLERS.get(&asset_type).map(|h| h.get_reader())`